src/borg/locking.py

```diff
@@ -158,6 +161,41 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_dead_lock(self):
+        for name in os.listdir(self.path):
+
+            try:
+                host_pid, thread = name.rsplit('-', 1)
+                host, pid = host_pid.rsplit('.', 1)
+            except ValueError:
+                # Malformed lock name? Or just soem new format we don't understand?
+                # It's safer to just exit
+                return False
+
+            if host != _hostname:
+                return False
+
+            try:
+                # This may not work in Windows.
+                os.kill(int(pid), 0)
```

Comment:
What if it takes ages for it to terminate?
I don't think you can wait for a process that you are not a parent of anyway.

```
     The wait4() call provides a more general interface for programs that need
     to wait for certain child processes, that need resource utilization sta-
     tistics accumulated by child processes, or that require options.  The
     other wait functions are implemented using wait4().
```

Note that kill(pid, 0) does not kill anything.
It just checks if the pid is a valid process or not (and that you can signals to it as well).